controller/worker/deployment/postgres.go

```diff
@@ -48,6 +49,10 @@ func (d *DeployJob) deployPostgres() (err error) {
 	if 2+len(state.Async) != d.Processes["postgres"] {
 		return loggedErr(fmt.Sprintf("pg cluster in unhealthy state (too few asyncs)"))
 	}
+	if reflect.DeepEqual(d.newReleaseState, d.Processes) {
```

Comment:
Don't use `reflect` if you don't have to. In this case it's a bit more code, but we can have a type-safe comparison.